conn.go

```diff
@@ -120,6 +128,13 @@ func NewConn(conn net.Conn, topic string, partition int) *Conn {
 	})
 }
 
+func transactionalIdToNullableString(transactionalId string) (result *string) {
```

Comment:
This seems more like a generic function to convert a `string` to a `*string`, how about a simpler name like
```suggestion
func nullableString(transactionalId string) (result *string) {
```
?

This is an implementation detail so feel free to keep it this way if you think it's more meaningful.

Reply:
This name was my initial intention. I was concerned that it had preconditions not reflected in the name. I think `emptyToNullable` should suit it.